### ISSUE TITLE
Router accepts text/plain requests which makes it CSRF vulnerable

### ISSUE THREAD

**Issue by lsieradzki** [ralscha/extdirectspring#150]
Hi,

This is the method from RouterController I am talking about :

`@RequestMapping(value = "/router", method = RequestMethod.POST, params = "!extAction")
	public void router(HttpServletRequest request, HttpServletResponse response,
Locale locale) throws IOException`

I wonder why it is not restricted to consume application/json only. The first line requires JSON anyway:
`		Object requestData = configurationService.getJsonHandler().readValue(request.getInputStream(), Object.class);
`

I was able to hit the services with text/plain with JSON in the body of the message from a different domain that the application is deployed to. CORS protection did not kick in because the request was in text/plain rather than application/json.

Thanks!

**Comment by ralscha** [comment]
Thanks for the report. 
I guess a consumes should solves that. Or what do you think?

```
@RequestMapping(value = "/router", method = RequestMethod.POST, params = "!extAction", consumes = "application/json")
```

**Comment by lsieradzki** [comment]
Yes, it should do the trick. However, it may break backward compatibility if someone relies on text/plain with JSON body.

**Comment by ralscha** [comment]
Not sure if that is a problem, because an application never contains user code that sends requests to these endpoints. An application always uses the ExtJs library and the library takes care of sending requests to these endpoints.

**Comment by ralscha** [comment]
Even ExtJs 3.4 sends request with the header. `content-type: application/json;charset=UTF-8`. 
Restricting the endpoints to json should not break any code.

**Comment by lsieradzki** [comment]
OK, so that is fine. Thanks for a quick reaction.

**Comment by ralscha** [comment]
Fix released with 1.8.1
